### DISCUSSION TITLE
Best way to bundle tnoodle in source code?

### DISCUSSION THREAD

**Discussion by xbjfk** [CubeLabsNZ/CubeTime#29] [Q&A] [answered]
So far I've considered:
- Download J2ObjC and TNoodle yourself, and put the includes and libraries in the 3rdparty directory manually (current solution)
  - Possibly automate with a script?
- Any form of including the Java or transpiled Objective C source code in CubeTime directly (including submodules):
  - I tried this initially and XCode was unhappy that there were files with the same basenames (eg Util.java in multiple packages)
- Just put the compiled libraries in this repo.
  - libjre_core.a is 204MB, does not seem to be possible.
  - Possibly use git LFS?
- Another option I am missing entirely?

**Comment by xbjfk** [comment, marked as answer]
https://github.com/CubeStuffs/tnoodle-lib-objc

**Reply by ghost** [reply]
Is this still the preferred method? It looks like a bit of a process, and It also looks like you've been doing some work behind the scenes since this was written.

**Reply by xbjfk** [reply]
Yes, I am doing some work on an alternative, hopefully to be finished by the end of this month. It will be faster, cross-platform native and easier to use.
